packages/apex-lsp-shared/src/types/priority.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the
+ * repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Priority levels for task scheduling and queue processing.
+ *
+ * Priority values:
+ * - Immediate (1): Critical tasks that must execute immediately
+ * - High (2): High-priority tasks
+ * - Normal (3): Standard priority tasks
+ * - Low (4): Low-priority tasks
+ * - Background (5): Background tasks
+ */
+export enum Priority {
```

Comment:
Were we trying to avoid enum use? I thought I recall something about that being a general rule of thumb because TS has no natural enum type. Personally I'm not super against it but I thought I saw elsewhere we changed to using strings so I'm a little mixed up on what the priority is supposed to be interpreted as.

Reply:
enumerations have numbers, which are more compact and comparisons more efficient.

Yes JS does not have this kind of enum, but string unions are strings.

If we get to the point of having to du any kind of in-memory compressions, enums gives us a head start.